src/bucket/BucketOutputIterator.cpp

```diff
@@ -67,17 +67,36 @@ BucketOutputIterator::put(BucketEntry const& e)
         ++mMergeCounters.mOutputIteratorTombstoneElisions;
         return;
     }
+    std::optional<BucketEntry> maybeInitEntry;
```

Comment:
I don't think you need to do this check here. If you move this to the bottom right before `*mBuf = ...`, you could get rid of the optional the complicated referencing later on and just say
```
if (shouldRewriteAsInit)
{
auto copy = e;
copy = init;
*mBuf = init;
}
else 
{
*mBuf = e;
}
```